### PR TITLE
skip special features in TestPodSecurityGAOnly

### LINKED ISSUE #116151: unstable map iteration order causes test to sometimes leave non-GA features disabled
### What happened?

PR https://github.com/kubernetes/kubernetes/pull/115324 was reverted by #116062 because it (erroneously) relied on OpenAPIV3 feature to be enabled, but `TestPodSecurityGAOnly` would non-deterministically leave the feature disabled after running.

I tracked it down to the fact that this test disables all `PreRelease` or `Beta` features in a loop which iterates over a `map`, which are known to have unstable iteration order:
https://github.com/kubernetes/kubernetes/blob/e55f2a9b54e0f6bd9ed12e50752d7d2545ab0cab/test/integration/auth/podsecurity_test.go#L74-L81

When key `AllBeta` is processed before key `OpenAPIV3`, then `AllBeta` would disable `OpenAPIV3` before `OpenAPIV3`s default value is to be saved.

Normally, `AllBeta` ignores features that have been explicitly set, so it only affects those which fall back on the `Default` value. But this issue does not just affect `OpenAPIV3` but any non-GA feature.

I suggest the loop in `TestPodSecurityGAOnly` be made to skip the special feature keys such as `AllBeta`.

/cc @liggitt 
/cc @apel

### What did you expect to happen?

The value before every feature gate should be restored after running `TestPodSecurityGAOnly`.

### How can we reproduce it (as minimally and precisely as possible)?

This test always fails on my machine (just pasted it into `test/integration/auth/podsecurity_test.go`):

```go
func TestRestoreFeatures(t *testing.T) {
	var targetFeature featuregate.Feature

	// Grab a beta feature form the list to test with
	for k, v := range utilfeature.DefaultMutableFeatureGate.GetAll() {
		if v.PreRelease == featuregate.Beta && v.Default {
			targetFeature = k
			break
		}
	}

	if len(targetFeature) == 0 {
		t.Log("no beta features to test against")
		return
	}

        // see if we iterate the feature map with AllBeta first within a few tries
	for i := 0; i < 10; i++ {
		attemptFunc := func() bool {
			fg := utilfeature.DefaultFeatureGate.DeepCopy()
			originalValue := fg.Enabled(targetFeature)

			func() {
				// Disable all alpha and beta features
				for k, v := range fg.DeepCopy().GetAll() {
					if v.PreRelease == featuregate.Alpha || v.PreRelease == featuregate.Beta {
						defer featuregatetesting.SetFeatureGateDuringTest(t, fg, k, false)()
					}
				}
			}()

			return originalValue != fg.Enabled(targetFeature)
		}

		if attemptFunc() {
			t.Fatal("original value for feature not same as new value after restoration")
			break
		}
	}
}
```

### Anything else we need to know?

_No response_

### Kubernetes version

<details>

```console
$ kubectl version
# paste output here
```

</details>

### Cloud provider

<details>

</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>

### Install tools

<details>

</details>

### Container runtime (CRI) and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

### PR DESCRIPTION
**Pull request by alexzielenski**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

Skips AllAlpha/AllBeta special features so that the list of features is properly restored for other tests.

#### Which issue(s) this PR fixes:

Fixes #116151

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

/sig auth
/assign @liggitt

### CODE REVIEW

File: test/integration/auth/podsecurity_test.go [hunk 1]
```diff
@@ -75,7 +75,11 @@ func TestPodSecurity(t *testing.T) {
 func TestPodSecurityGAOnly(t *testing.T) {
 	// Disable all alpha and beta features
 	for k, v := range utilfeature.DefaultFeatureGate.DeepCopy().GetAll() {
-		if v.PreRelease == featuregate.Alpha || v.PreRelease == featuregate.Beta {
+		if k == "AllAlpha" || k == "AllBeta" {
```
Comment (alexzielenski) [inline test/integration/auth/podsecurity_test.go:78]: also considered just swapping out the entire `utilfeature.DefaultFeatureGate` but that cannot be done atomically, not sure if that is a consideration in the test environment

Comment (liggitt) [reply test/integration/auth/podsecurity_test.go:78]: this is fine for now, I'll open a follow up to make SetFeatureGateDuringTest safe no matter what you throw at it

**Comment by liggitt** [comment]
/hold for the general fixup in https://github.com/kubernetes/kubernetes/pull/116154
if this goes green and that fails CI, feel free to merge this early and I'll clean it up after the other merges

**Pull request merged**